test/app.test.ts

```diff
@@ -411,4 +439,35 @@ describe('Test app.ts', () => {
         expect(res.text).toContain(t('errors.problem'));
         expect(res.text).toContain(t('errors.dataset_missing'));
     });
+
+    test('Dataset preview is rendered in the frontend', async () => {
+        const cookies = await setDatasetToSession();
+        const res = await request(app)
+            .get('/en-GB/publish/preview')
+            .set('Cookie', cookies)
+            .set('User-Agent', 'supertest');
+
+        expect(res.status).toBe(200);
+        // Header
+        expect(res.text).toContain(`<th scope="col" class="govuk-table__header">
+                                        ID
+                                    </th>`);
+        expect(res.text).toContain(`<th scope="col" class="govuk-table__header">
+                                        Text
+                                    </th>`);
+        expect(res.text).toContain(`<th scope="col" class="govuk-table__header">
+                                        Number
+                                    </th>`);
+        // First Row
+        expect(res.text).toContain(`<td class="govuk-table__cell">1</td>`);
+        expect(res.text).toContain(`<td class="govuk-table__cell">test1</td>`);
+        expect(res.text).toContain(`<td class="govuk-table__cell">3423196</td>`);
+        expect(res.text).toContain(`<td class="govuk-table__cell">2001-09-20</td>`);
+        // Last Row
+        expect(res.text).toContain(`<td class="govuk-table__cell">2</td>`);
+        expect(res.text).toContain(`<td class="govuk-table__cell">AcHVoWJblA</td>`);
+        expect(res.text).toContain(`<td class="govuk-table__cell">4470652</td>`);
+        expect(res.text).toContain(`<td class="govuk-table__cell">2002-03-18</td>`);
+        expect(res.text).toContain('Showing rows 1 - 2 of 2');
+    });
```

Comment:
I'd probably avoid testing the exact html that comes back and just make sure the data is present - otherwise if we change the classes etc it will break the test.

Reply:
We can resolve this in another pass though. :ship: